nixos/modules/services/games/asf.nix

```diff
@@ -0,0 +1,200 @@
+      type = types.attrsOf (types.submodule {
+        options = {
+          username = mkOption {
+            type = types.str;
+            description =
+              "Name of the user to log in. Default is attribute name.";
+            default = "";
+          };
+          passwordFile = mkOption {
+            type = types.path;
+            description =
+              "Path to a file containig the password. The file must be readable by the <literal>asf</literal> user/group.";
+          };
+          enabled = mkOption {
+            type = types.bool;
+            default = true;
+            description = "Whether to enable the bot on startup.";
+          };
+          settings = mkOption {
+            type = types.attrs;
+            description =
+              "Additional settings that are documented here: https://github.com/JustArchiNET/ArchiSteamFarm/wiki/Configuration#bot-config";
+            default = { };
+          };
+        };
+      });
+      description = ''
+        Bots name and configuration
+      '';
+      example = {
+        exampleBot = {
+          username = "alice";
+          passwordFile = "/my/super/secret/password";
+          settings = { SteamParentalCode = "1234"; };
+        };
+      };
+      default = { };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users = {
+      users.asf = {
+        home = cfg.dataDir;
+        createHome = true;
+        isSystemUser = true;
+        group = "asf";
+        description = "Archis-Steam-Farm service user";
+      };
+      groups.asf = { };
+    };
+
+    systemd.services = {
+      asf = {
+        description = "Archis-Steam-Farm Service";
+        after = [ "network.target" ];
+
+        serviceConfig = {
+          User = "asf";
+          Group = "asf";
+          WorkingDirectory = cfg.dataDir;
+          Type = "simple";
+          ExecStart =
+            "${cfg.package}/bin/ArchiSteamFarm --path ${cfg.dataDir} --process-required --no-restart --service ";
+
+          # mostly copied from the default systemd service
+          PrivateTmp = true;
+          LockPersonality = true;
+          PrivateDevices = true;
+          PrivateIPC = true;
+          PrivateMounts = true;
+          PrivateUsers = true;
+          ProtectClock = true;
+          ProtectControlGroups = true;
+          ProtectHostname = true;
+          ProtectKernelLogs = true;
+          ProtectKernelModules = true;
+          ProtectKernelTunables = true;
+          ProtectProc = "invisible";
+          ProtectSystem = "full";
+          RemoveIPC = true;
+          RestrictAddressFamilies = "AF_INET AF_INET6";
+          RestrictNamespaces = true;
+          RestrictRealtime = true;
+          RestrictSUIDSGID = true;
+        };
+
+        preStart = ''
+          set -x
+          # we remove config to have no complexity when creating the required files/directories
+          mkdir config || true
+          rm -rf config/*.json
+          rm -rf www
```

Comment:
Is placing configuration files in an immutable directory (the nix store) not an option?

Reply:
How can I get the real name of a file in the nix store at runtime? Having bots that are named <sha>-name-of-bot is a bit awkward
This is what I currently changed: http://0x0.st/-hxH.txt

```
# ls config/
7jhdd88snd25m2n2pz8l434a1nlqd2nf-acc1.json  ASF.json
ASF.db                                                  l6fjn0ph5ars4m3vvv6nxskdcw0n070w-acc2.json
```
And this what my config directory looks like

Reply:
I think you want $out to be a directory and place the json file there with whatever name you want.

Reply:
How can I have $out from a module?

Reply:
Î think we would need to build a derivation for that.

Reply:
A separate derivation that takes in all the files as an argument (which i change by using .override?) and then I link `lib/asf/bots` (or something similar) to the correct dir...

Does that sound ok?